README.md

```diff
@@ -1,339 +1,147 @@
-  id: 'todos',
-  sync: { /* sync configuration */ },
-  schema: /* optional schema */
-});
-
-// Create a mutation
-const mutation = useOptimisticMutation({
-  mutationFn: async ({ mutations }) => {
-    // Implement your mutation logic here
-    // This function is called when mutations are committed
-  }
-});
-
-// Use the mutation with collection operations
-mutation.mutate(() => {
-  insert({ text: 'New todo' });
-});
-```
-
-Returns:
-
-- `data`: An array of all items in the collection
-- `state`: A Map containing all items in the collection with their internal keys
-- `insert`: Function to add new items to the collection
-- `update`: Function to modify existing items
-- `delete`: Function to remove items from the collection
+## 💥 Usage example
 
-#### `preloadCollection`
+Sync data into collections. Bind live queries to your components. Make writes using transactional mutations.
 
-Preloads data for a collection before rendering components.
+```tsx
+import { Collection, createTransaction, useLiveQuery } from '@tanstack/react-optimistic'
+import type { MutationFn, PendingMutation } from '@tanstack/react-optimistic'
 
-```typescript
-await preloadCollection({
+// Sync data into collections
+export const todoCollection = new Collection<Todo>({
   id: 'todos',
-  sync: { /* sync configuration */ },
-  mutationFn: { /* mutation functions */ },
-  schema: /* optional schema */
-});
-```
-
-Features:
-
-1. Returns a promise that resolves when the first sync commit is complete
-2. Shares the same collection instance with `useCollection`
-3. Handles already-loaded collections by returning immediately
-4. Avoids duplicate initialization when called multiple times with the same ID
-
-### Data Operations
-
-#### Insert
-
-```typescript
-// Insert a single item
-insert({ text: "Buy groceries", completed: false })
-
-// Insert multiple items
-insert([
-  { text: "Buy groceries", completed: false },
-  { text: "Walk dog", completed: false },
-])
-
-// Insert with custom key
-insert({ text: "Buy groceries" }, { key: "grocery-task" })
-```
-
-#### Update
+  sync: {
+    // use a sync engine like ElectricSQL or any data
+    // loading strategy including TanStack Query
+  },
+  schema: todoSchema // standard schema interface
+})
 
-We use a proxy to capture updates as immutable draft optimistic updates.
+// Bind live queries to your components
+const Todos = () => {
+  const { data: todos } = useLiveQuery(q =>
+    q // queries support joins, aggregates, etc.
+      .from({ todoCollection })
+      .select('@t.id', '@t.text', '@t.completed')
+      .keyBy('@id')
+  )
 
-```typescript
-// Update a single item
-update(todo, (draft) => {
-  draft.completed = true
-})
+  return <List items={todos} />
+}
 
-// Update multiple items
-update([todo1, todo2], (drafts) => {
-  drafts.forEach((draft) => {
-    draft.completed = true
+// Make writes using transactional mutations
+const tx = createTransaction({ mutationFn: /* ... */ })
```

Comment:
Let's use useOptimisticMutation

Reply:
I just fleshed out a bit and pushed some tweaks.

What's the best way to approach the usage example here and the longer usage docs -- could you do a pass with your preferred APIs? Or want to give me some pointers and I'll try and dig out of the tests and react/todo/example?

Reply:
The tests could be enough to get you going but I'm happy to pick it up Monday too

Reply:
I updated the README usage example.